scripts/functions

```diff
@@ -1183,6 +1183,14 @@ CT_DoConfigSub() {
     fi
 }
 
+# Normally, each step is executed in a sub-shell and thus cannot modify the
+# environment for the next step(s). When this is needed, it can do so by
+# invoking this function.
+# Usage: CT_EnvModify VAR VALUE
+CT_EnvModify() {
+    echo "${1}=\"${2}\"" >> "${CT_BUILD_DIR}/env.modify.sh"
+}
```

Comment:
I think a better solution would have been something like:

    # Store extra environment for later steps
    CT_StoreEnv() {
        printf "%s='%s'\n" "${1}" "${2}" >>"${CT_BUILD_DIR}/extra-env.sh"
    }

    # Load extra environment from previous steps
    CT_LoadEnv() {
        if [ -f "${CT_BUILD_DIR}/extra-env.sh" ]; then
            . "${CT_BUILD_DIR}/extra-env.sh"
        fi
    }

And then you use CT_LoadEnv in the top-level sub-shell:

    scripts/crosstool-NG.sh.in:
    623        if [ ${do_it} -eq 1 ]; then
    624            ( CT_LoadEnv; do_${step} )

By loading the environment in the sub-shell, you won;t pollute the
top-level environment, while still allowing passing environment from
steps to steps.

Note that the environment file is not removed after being parsed; it is
cumulative: steps can add new environment, valid for the next steps.